pkg/querier/iterators/chunk_merge_iterator.go

```diff
@@ -183,21 +183,27 @@ func (c *chunkMergeIterator) At() (t int64, v float64) {
 	return c.currTime, c.currValue
 }
 
-func (c *chunkMergeIterator) AtHistogram() (int64, *histogram.Histogram) {
+func (c *chunkMergeIterator) AtHistogram(h *histogram.Histogram) (int64, *histogram.Histogram) {
 	if c.currValueType != chunkenc.ValHistogram {
 		panic(fmt.Errorf("chunkMergeIterator: calling AtHistogram when cursor is at different type %v", c.currValueType))
 	}
-	return c.currTime, c.currHistogram
+	if h != nil {
+		c.currHistogram.CopyTo(h)
+	}
+	return c.currTime, c.currHistogram.Copy()
 }
 
-func (c *chunkMergeIterator) AtFloatHistogram() (int64, *histogram.FloatHistogram) {
+func (c *chunkMergeIterator) AtFloatHistogram(fh *histogram.FloatHistogram) (int64, *histogram.FloatHistogram) {
 	if c.currValueType == chunkenc.ValHistogram {
 		return c.currTime, c.currHistogram.ToFloat(nil)
 	}
 	if c.currValueType != chunkenc.ValFloatHistogram {
 		panic(fmt.Errorf("chunkMergeIterator: calling AtFloatHistogram when cursor is at different type %v", c.currValueType))
 	}
-	return c.currTime, c.currFloatHistogram
+	if fh != nil {
+		c.currFloatHistogram.CopyTo(fh)
```

Comment:
Should you return here? Aren't you supposed to avoid making a copy if h is non-nil?

Reply:
removed code, was only used in tests anyway, no point in trying to optimize